### ISSUE TITLE
Bookmark Duplication

### ISSUE THREAD

**Issue by Soben** [brave/brave-browser#6461]
## Description 
I keep my Bookmarks synced using brave://sync on a work laptop, personal laptop, and my iPhone.

About a week or so ago, I started noticing a bookmark in my toolbar that kept doubling up, and last week the bar started reorganizing the bookmarks placing the new duplications at the top of the list.

Today is my first time on my personal laptop extensively in a while, and I'm noticing a LOT of duplicated items throughout my bookmarks, including multiple instances of some bookmarks beyond the ones I first noticed.

## Steps to Reproduce
I'm not entirely sure how to replicate besides to have multiple devices syncing.

## Actual result:
https://cl.ly/4585cd7939cd

## Expected result:
Bookmarks, especially ones that were in place before even implementing the brave://sync feature (which are most of mine), to not duplicate.

## Reproduces how often: 
intermittent, have not been able to determine a solid cause yet.

## Brave version (brave://version info)

Brave | 0.69.132 Chromium: 77.0.3865.90 (Official Build) (64-bit)
-- | --
Revision | 58c425ba843df2918d9d4b409331972646c393dd-refs/branch-heads/3865@{#830}
OS | macOS Version 10.15 (Build 19A583)

Same on work laptop except the OS version is 10.14

## Version/Channel Information:
Have not tested this on any additional channels.

## Other Additional Information:
Oddly enough, this issue did seemingly begin after I started Brave Rewards, but I have not disabled the feature as I do want to continue to participate in the program. I also am wary of making changes to my bookmarks as I don't want to completely lose some.

## Miscellaneous Information:
N/A

**Comment by mauroservienti** [comment]
I'm having the exact same problem on Windows 10 machines (a workstation and a laptop) and 2 iOS devices (iPhone/iPad).
What I observe is that the workstation bookmarks are duplicated multiple times from laptop ones. Here is what I see trimmed down to a 1 single bookmark scenario/sample:

- workstation has 2 bookmarks (site a and site b)
- laptop has 1 bookmark (site a)
- after a sync
  - workstation gets to (site a, site b, site a)
  - laptop gets to (site a, site b)
- after another sync
  - workstation gets to (site a, site b, site a, site b, site a)
  - laptop gets to (site a, site b, site a, site b)

they constantly grow and are never de-duplicated. It seems that the workstation, that is the machine from which I created the sync chain is the only one that never gets updated., it gets only duplicates. So a change to laptop is propagated to workstation as a duplicated and not as an update.

**Comment by mpaluta** [comment]
I've experienced this for months now. Bookmarks duplicate, move, disappear, reappear later. Ordering of bookmarks is inconsistent among all devices as well. I sync Windows, Linux, and iOS in my chain, but I suspect device type doesn't matter.

**Comment by brimwats** [comment]
I have also been experiencing this for months. Not only did one laptop have its bookmarks duplicated 8x, I lost lost a valuable folder of research for my PhD because of syncing issues. It's gotten to the point where I've had to turn off syncing. Thankfully, I managed to back up the folder last week, but it was incredibly alarming, and really disappointing. I really like brave but I am starting to investigate other options as this is a mess.

**Comment by mauroservienti** [comment]
Any update on this one?

**Comment by mauroservienti** [comment]
To be honest, this is beyond acceptable:

![image](https://user-images.githubusercontent.com/1325611/73289690-6f6a7300-41fd-11ea-8388-5037275b31b2.png)

(not to mention the mess in each folder)

I spend more time fixing those horrible sync problems than working.

**Comment by rafaces** [comment]
Same problem here;
I have synced bookmarks between 2 Macbooks, 1 iPad and 1 Android phone;
The bookmark list is a mess, and although I try to delete duplicates every other day, the list grows all the time and I can't handle manually fixing it every time anymore

**Comment by masparrow** [comment]
Some additional info here (quite a bit so bear with me). I repeatedly get duplicate bookmarks, however I have tracked it down and stopped it from happening in my instance (I'm not saying it's the same issue as others are seeing).

I have multiple machines, mobile devices and OS's all running Brave and I want to sync across all of them. 

With clean installs, as soon as I set up the _existing_ sync chain I have from previous installs, I start getting duplicates across all the devices within minutes of configuring them.

However, I found that my Android phone was on a _different sync chain_ with a different set of key words. I have never set this up to be on a different sync chain - however, I cannot rule out 'user error' - though I'm 80% certain it was nothing to do with me. I've only ever set up sync once, and only ever lifted one set of key words.

My main development PC was on _both_ of these sync chains some how. 

So even though I'd done a clean install on my dev PC, and then only added it to my 'correct' sync chain, somehow it was crossing over to the 'other' sync chain (presumably keying off the PC's device name), and I believe that's when duplicate bookmark shenanigans were initiated. 

It might be worth those affected by this just checking each device they have (ever) installed Brave on, and seeing what sync chain key words they offer to add a new device, and comparing them to their other devices. 

Removing both devices in the 'other' sync chain, and then only using the correct sync chain seems to have done the trick for me, and I have 3 x Windows, 3 x OSX, 1 x iOS and 1 x Android installs all with the same single set of bookmarks, running for about a week without any duplicates. @Brave this is using the latest official release builds on all platforms.
